### PR TITLE
HDE 464 enable automapper to handle no keys

### PR DESCRIPTION
**Pull request by imranq2**

### CODE REVIEW

File: spark_auto_mapper/automappers/automapper.py [hunk 1]
```diff
@@ -41,11 +43,21 @@ def transform_with_data_frame(
...
     def transform(self, df: DataFrame) -> DataFrame:
-        assert self.keys and len(self.keys) > 0
         # if source_view is specified then load that else assume that df is the source view
         source_df: DataFrame = df.sql_ctx.table(
             self.source_view
         ) if self.source_view else df
+        # if keys are not specified then add a __row_id column to both the source and the destination
+        #   and use that as key
+        if not self.keys or len(self.keys) == 0:
+            assert not self.view or not SparkHelpers.spark_table_exists(
```
Comment (adhearn) [inline spark_auto_mapper/automappers/automapper.py:53]: Is this assert required? I can think of a case where we want to rename columns in a view, so `self.view` and `self.source_view` would refer to the same view and `self.keys` would be `None`. Is that type of view manipulation discouraged?

**Review by adhearn — COMMENTED** [review]
This ended up quite a bit cleaner than I thought it might. There's one spot where I'm confused, and I want to make sure I understand it fully before approving, but otherwise I'm pretty happy with how this turned out.

**Comment by kpaul** [comment]
> This ended up quite a bit cleaner than I thought it might. There's one spot where I'm confused, and I want to make sure I understand it fully before approving, but otherwise I'm pretty happy with how this turned out.

This looks good to me outside of the question that @adhearn has.

File: spark_auto_mapper/automappers/automapper.py (hunk 1 above)
Comment (imranq2) [reply spark_auto_mapper/automappers/automapper.py:53]: If we create new ids and the source table already has ids then they will not match.  If you’re adding yo an existing table in memory then you should have keys in both tables

**Review by michaelvidal24 — APPROVED** [review]

**Pull request merged**